.github/quarkusbuilditemdoc.java

```diff
@@ -0,0 +1,167 @@
+//usr/bin/env jbang "$0" "$@" ; exit $?
+//DEPS info.picocli:picocli:4.5.0
+//DEPS org.jboss.forge.roaster:roaster-jdt:2.22.2.Final
+//DEPS org.eclipse.collections:eclipse-collections:10.4.0
+//DEPS org.yaml:snakeyaml:1.27
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.concurrent.Callable;
+
+import org.eclipse.collections.api.multimap.Multimap;
+import org.eclipse.collections.api.multimap.MutableMultimap;
+import org.eclipse.collections.api.tuple.Pair;
+import org.eclipse.collections.impl.factory.Multimaps;
+import org.eclipse.collections.impl.tuple.Tuples;
+import org.jboss.forge.roaster.Roaster;
+import org.jboss.forge.roaster.model.source.JavaClassSource;
+import org.yaml.snakeyaml.Yaml;
+import picocli.CommandLine;
+import picocli.CommandLine.Command;
+
+@Command(name = "builditemdoc", mixinStandardHelpOptions = true, version = "builditemdoc 0.1",
+        description = "builditemdoc made with jbang")
+class quarkusbuilditemdoc implements Callable<Integer> {
+
+    public static void main(String... args) {
+        int exitCode = new CommandLine(new quarkusbuilditemdoc()).execute(args);
+        System.exit(exitCode);
+    }
+
+    // find extensions -name "*BuildItem.java" | jbang ~/workspace/quarkus-jbang-catalog/builditemdoc.java > docs/src/main/asciidoc/build-items.adoc
+    @Override
+    public Integer call() throws Exception {
+        final Multimap<String, Pair<Path, JavaClassSource>> multimap = collect();
+        Map<String, String> names = extractNames(Paths.get("."), multimap.keySet());
+        printDocumentHeader();
+        names.forEach((key, name) -> {
+            printTableHeader(name);
+            for (Pair<Path, JavaClassSource> source : multimap.get(key)) {
+                printTableRow(source);
+            }
+            printTableFooter();
+        });
+        return 0;
+    }
+
+    private String buildDescriptionFromJavaDoc(JavaClassSource source) {
+        if (!source.hasJavaDoc()) {
+            return "<i>No Javadoc found</i>";
+        }
+        return source.getJavaDoc().getText();
+    }
+
+    private Multimap<String, Pair<Path, JavaClassSource>> collect() throws IOException {
+        MutableMultimap<String, Pair<Path, JavaClassSource>> multimap = Multimaps.mutable.list.empty();
+        String line;
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
+            while ((line = br.readLine()) != null) {
+                Path path = Paths.get(line);
+                JavaClassSource source = Roaster.parse(JavaClassSource.class, path.toFile());
```

Comment:
Does this only list public classes ?
If a build item is not public, it's not designed to be used outside the extension